.github/workflows/build_and_test.yml

```
@@ -198,6 +198,19 @@ jobs:
         ports:
           - 9411:9411
 
+      redis-auth:
+        image: redis/redis-stack-server
+        # Set health checks to wait until redis has started
+        options: >-
+          --health-cmd "redis-cli ping"
```

```suggestion
          --health-cmd "redis-cli -a passdefault ping"
```

As it requires password, we may need to pass it when doing healthcheck.